questions.ts

```diff
@@ -625,182 +642,229 @@ export const askQuestions = async (
+              prefix: baseUrl,
+              depth: 1,
+            });
+            if (shouldContinue !== undefined && !shouldContinue) {
+              break;
+            }
+          }
+          program.dataSources.push(dataSource);
+        }
+
+        // No need to ask for another data source if user selected example data
+        if (selectedSource === "exampleFile") {
+          break;
         }
       }
+
+      if (
+        program.dataSources.length === 0 ||
+        program.dataSources[0].type === "none"
+      ) {
+        program.engine = "simple";
+      } else {
+        program.engine = "context";
+      }
     }
-  } else if (!program.dataSource) {
+  } else if (!program.dataSources) {
     // Handle a case when engine is specified but dataSource is not
     if (program.engine === "context") {
-      program.dataSource = {
-        type: "folder",
-        config: {},
-      };
+      program.dataSources = [
+        {
+          type: "folder",
+          config: {},
+        },
+      ];
     } else if (program.engine === "simple") {
-      program.dataSource = {
-        type: "none",
-        config: {},
-      };
+      program.dataSources = [
+        {
+          type: "none",
+          config: {},
+        },
+      ];
     }
   }
 
   if (
-    program.dataSource?.type === "file" ||
-    program.dataSource?.type === "folder"
+    program.dataSources.some(
+      (ds) => ds.type === "file" || ds.type === "folder",
+    ) &&
+    !program.useLlamaParse
   ) {
     if (ciInfo.isCI) {
+      program.useLlamaParse = getPrefOrDefault("useLlamaParse");
       program.llamaCloudKey = getPrefOrDefault("llamaCloudKey");
     } else {
-      const dataSourceConfig = program.dataSource.config as FileSourceConfig;
-      dataSourceConfig.useLlamaParse = program.llamaParse;
-
-      // Is pdf file selected as data source or is it a folder data source
-      const askingLlamaParse =
-        dataSourceConfig.useLlamaParse === undefined &&
-        (program.dataSource.type === "folder" ||
-          (program.dataSource.type === "file" &&
-            dataSourceConfig.paths?.some((p) => path.extname(p) === ".pdf")));
-
-      // Ask if user wants to use LlamaParse
-      if (askingLlamaParse) {
-        const { useLlamaParse } = await prompts(
-          {
-            type: "toggle",
-            name: "useLlamaParse",
-            message:
-              "Would you like to use LlamaParse (improved parser for RAG - requires API key)?",
-            initial: true,
-            active: "yes",
-            inactive: "no",
-          },
-          handlers,
+      if (!program.useLlamaParse && program.engine === "context") {
+        const askingLlamaParse = program.dataSources.some(
+          (ds) =>
+            ds.type === "folder" ||
+            (ds.type === "file" &&
+              (ds.config as FileSourceConfig).paths?.some(
+                (p) => path.extname(p) === ".pdf",
+              )),
```

Comment:
i think we can simplify this and always ask for llamaparse if we select file or folder